flask_hw/flask_rest/api/room/rooms.py

```diff
@@ -0,0 +1,118 @@
+import logging
+
+from flask_restful import Resource, marshal_with, marshal
+
+from api.room.room_parsers import data_valid_for
+from api.room.structure import room_structure
+from db import Rooms, db
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+class RoomsRes(Resource):
+
+    @marshal_with(room_structure)
+    def get(self, number=None):
+        room_all = Rooms.query.all()
+        header = {
+            "Cache-Control": "no-store, no-cache, must-revalidate",
+            "Pragma": "no-cache"
+        }
+        args = data_valid_for('GET')
+        logging.debug(room_all)
+        logging.debug(number)
+        logging.debug(args)
+        if number:
+            room = Rooms.query.get(number)
+            if room:
+                return room, header
+            else:
+                # todo: how to display message instead of Rooms
+                #  structure with nulls?
+                return {"message": "room not found"}, 404, header
+        else:
+            if args['status']:
+                rooms = Rooms.query.filter_by(status=args['status']).all()
+                return rooms, header
+            # todo: add if status not found?
```

Comment:
add

Reply:
added code 404 status, but the message is suppressed by the marshal's null output